src/filter.rs

```diff
@@ -18,11 +19,12 @@ impl OtelFilter {
 
 impl Default for OtelFilter {
     fn default() -> Self {
-        Self::new(crate::otlp::read_log_level_from_env())
+        Self::new(read_otel_log_level_from_env())
     }
 }
 
 impl<S: Subscriber> Filter<S> for OtelFilter {
+    #[inline]
     fn enabled(&self, meta: &Metadata<'_>, _: &Context<'_, S>) -> bool {
```

Comment:
This method should never be called, so there is little sense in inlining it.

`tracing` crate is highly optimised. It will call `callsite_enabled` exactly once per callsite (i.e. per unique tracing call in the code) and then cache it's results.

It could use `enabled` to perform dynamic matching for each span based on the context, but our current `callsite_enabled` implementation prevents it.